### ISSUE TITLE
WOW Animations don,t work when i scroll

### ISSUE THREAD

**Issue by ali-hasani** [tinesoft/ngx-wow#16]
Hi 
I have no problem with wow, but animations only work when the page loads for first time, and when i scroll page animations don't work ..
Is there any thing that I haven't done yet ?

**Comment by tinesoft** [comment]
Hi @alihasaniGit ,

By default, when WOW is intialialized, animations are only run once, when you scroll to items, as you can [see](http://mynameismatthieu.com/WOW/) on official website.

But you can easily re-init the library though,  by calling `NgwWowService.init()`
i suggest adding this in your app main component, when done navigating to a page:

```
import { Component, OnInit, OnDestroy } from '@angular/core';
import { Router, NavigationEnd } from '@angular/router';
import { NgwWowService } from 'ngx-wow';

@Component({
  selector: 'app-root',
  templateUrl: './app.component.html',
  styleUrls: ['./app.component.scss']
})
export class AppComponent implements OnInit, OnDestroy {

  constructor(private router: Router, private wowService: NgwWowService){
    this.router.events.filter(event => event instanceof NavigationEnd).subscribe(event => {
      // Reload WoW animations when done navigating to page,
      // but you are free to call it whenever/wherever you like
      this.wowService.init(); 
    });
}
```

**Comment by alihasaniGit** [comment]
I have done your suggest. actually i put "NgwWowService" in my "HomePageComponent" (instead of appComponent) and i initialized wowService in the component's constructor as you said. totally i did all things that you said, but now when i scroll page, animations don't work, whereas they work at first page load ! (at the first time page loads)

Furthermore I think you have another mistake in your documentation in the section you have said "list wowjs in angular-cli as a global library". i mean this section :
```
   "scripts": [
        "../node_modules/wowjs/dist/wowjs.js"
      ],
```
When I got error and checked, I saw there is no "wowjs.js' file in "dist" folder. As you know it's name is **wow.js**. if I say correct, fix it ..

I waiting for your solution for animation problem ..

Thanks a lot friend ..

**Comment by tinesoft** [comment]
Hi,

I fixed the typo in the README.md (thanks again for reporting 👍 ).

Your issue with "animations only working at first page load" is difficult to diagnose.
Do you have a minimal repro project that i could checkout and try locally?

**Comment by alihasaniGit** [comment]
Please fix the typo problem in your [documentation](https://tinesoft.github.io/ngx-wow/doc/index.html) too.

Please wait. I want to try to fix problem. I'll let you know if i can't ..

**Comment by alihasaniGit** [comment]
Hi friend 

I couldn't solve my problem, but I bring some information for you to get help.
The information I want to give are associated with some questions.

First one : 
I put the pieces of code (you said in docs) this way :

![2](https://user-images.githubusercontent.com/29506320/34459953-f059a010-ee14-11e7-952b-fd26a0af8207.PNG)

Second one :
What should I do in this section? do I have to do any thing?

![1](https://user-images.githubusercontent.com/29506320/34459925-214dbe64-ee14-11e7-9c77-ed525e002201.PNG)

Third one :
As you saw in "Second one" section I put a debugger in the function but when an element reveals in the page debugger isn't called .. 

Forth one :
When I give "wow fadeIn" class to an element, when I scroll to that, it's visibility is "hidden" and when I inspect (Chrome Console) on the element this is the view that I see : 

![3](https://user-images.githubusercontent.com/29506320/34460006-aa0d2152-ee16-11e7-99be-64bdd76ab604.PNG)

Thanks for help ..

**Comment by tinesoft** [comment]
Hi @alihasaniGit ,

Once again, sorry because the documentation was missing another important info: referencing `animate.css` (which is installed and internally used by `wowjs`to actually animate items) in your project.

This can be achieved by either:

* editing `angular-cli.json` as such:
```
      "styles": [
        "../node_modules/animated.css/animate.css"
      ],
```
* or by importing in your main `styles.scss` (or `styles.sass`, `styles.less`, `styles.styl`) file as such:
```sass
...
@import "~animate.css/animate.css";
...
```

I have updated the docs accordingly an published v1.0.2 to NPM.

Also i clarified the [rest of the documentation](https://github.com/tinesoft/ngx-wow/blob/master/README.md#usage) regarding your question on `wowjs`' Subscription (question 2): you don't need to subscribe to `this.wowService.itemRevealed$` at all, for the library to work. I only mention that in the docs so that you know you can be notified when an item is revealed by `wowjs` in case this information might be useful for your use case.

**Comment by alihasaniGit** [comment]
Unfortunately I have the problem yet !
Did you notice to the "Forth one" section in the previous comment ?
Didn't it give you any clue ?

**Comment by tinesoft** [comment]
Hi,

You 4th question didn't help much, and i thought it would have been fixed anyway.

Can you confirm that `animate.css` is correctly loaded for your app? 
For example if you have in your dom:
```
 <div class="col wow bounce">
        <p class="circle bg-yellow">aint joke</p>
 </div>
```
Make sure that if you inspect the corresponding node in Chrome Dev Tools, you have the style `.bounce` defined.

![image](https://user-images.githubusercontent.com/4053092/34494789-4e0b2a2e-eff2-11e7-8e92-79ccc4900649.png)

**Comment by alihasaniGit** [comment]
Hi friend,

It,s loads correctly, but as you see some CSS styles (style attribute in the tag) overwrites '.bounce' class styles.

![2-min](https://user-images.githubusercontent.com/29506320/34639455-ef99df9c-f2f5-11e7-9584-5890a546822c.PNG)

I don't know why !

**Comment by tinesoft** [comment]
Hi @alihasaniGit 

At this point, it is very difficult for me to identify where the problem might be without seeing the source code...
Can you push your repository? or at least create  a minimal project that reproduces the issue?

**Comment by rituagrawal904** [comment]
I am also facing the same problem. when I am scrolling down, elements are still hidden and animation name still none. Please provide solution for this 
Thanks

![image](https://user-images.githubusercontent.com/26505016/34723591-37ecc9a8-f571-11e7-870e-d4ec0f607b4c.png)

**Comment by tinesoft** [comment]
Hi @rituagrawal904 

I'm more than willing to fix the issue too, but without further context i'm stuck as well.
Currently, i cannot even reproduce the problem locally, so a minimal repo project will help a lot

**Comment by rituagrawal904** [comment]
Can we initiate wow.js on element scrolling instead of window scroll. I am using angular2 and winow is steady it is not scrolling and I want to animate elements on page scroll

**Comment by henrykalex** [comment]
I think for someone the solution is to add the css class: (global)
.wow {
   visibility: hidden;
}
Reference: https://stackoverflow.com/questions/23531064/wow-js-happens-on-page-load-and-not-on-scroll

**Comment by tinesoft** [comment]
Thanks for your input @henrykalex 👍 
@alihasaniGit, @rituagrawal904 :  can you confirm it fixes your issue?

**Comment by alihasaniGit** [comment]
I have the problem yet, but thanks for your solution ..

**Comment by mateusneves** [comment]
I am with the same problem

**Comment by mateusneves** [comment]
My project uses the fullpage.js script. My solution for the wow script work in scroll page was change the option scrollBar for true in fullpage options.

**Comment by airammateos** [comment]
Hi,

Just to leave a comment saying that I have to import and init WOW as said in this post:
https://stackoverflow.com/questions/44454855/how-to-setup-wow-js-in-angular-2-and-4

**Comment by ReevMich** [comment]
> I had the same issue, turned out it was because the window scroll event wasnt firing because I was scrolling inside a container div. The solution is designating the `scrollContainer` in the init options.
> 
> The types for `NgwWowConfig` are incorrect, it lists `scrollContainer` as an `HTMLElement` but it is actually a css selector so should be `string`.
> 
> My working config looks like this
> 
> ```
> this.wowService.init({
>     scrollContainer: '#scrollContainer' as any
> });
> ```

This is correct will there be a PR that address this?

**Comment by tinesoft** [comment]
Hi all, 

I fixed the typing issue and released [v2.0.1](https://github.com/tinesoft/ngx-wow/releases/tag/v2.0.1).

Thanks @attbgold for reporting

**Comment by D0ubleTrouble** [comment]
Thanks @mateusneves your solution saved my day.
